// @letanure/resend-cli/dist/src/utils/output.js
// Output formatting utilities for CLI commands
// Output success result
export function outputSuccess(data, format, textCallback) {
    if (format === 'json') {
        const response = {
            success: true,
            data,
        };
        console.log(JSON.stringify(response, null, 2));
    }
    else {
        textCallback();
    }
}
// Output validation errors
export function outputValidationErrors(errors, format, textCallback) {
    if (format === 'json') {
        const response = {
            success: false,
            error: 'We found an error with one or more fields in the request.',
            code: 'validation_error',
            errors: errors.map((err) => ({
                field: String(err.path),
                message: err.message,
            })),
        };
        console.log(JSON.stringify(response, null, 2));
    }
    else {
        textCallback();
    }
}
// Output general error
export function outputError(message, format, textCallback) {
    if (format === 'json') {
        const response = {
            success: false,
            error: message,
        };
        console.log(JSON.stringify(response, null, 2));
    }
    else {
        textCallback();
    }
}
