// @letanure/resend-cli/dist/src/modules/api-keys/Menu.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Layout } from '../../components/ui/layout.js';
import { Menu as UIMenu } from '../../components/ui/Menu.js';
import { config } from '../../config/config.js';
export var MenuState;
(function (MenuState) {
    MenuState["create"] = "create";
    MenuState["list"] = "list";
    MenuState["delete"] = "delete";
})(MenuState || (MenuState = {}));
const MAIN_MENU_ITEMS = [
    {
        id: MenuState.create,
        label: 'Create',
        description: 'Create a new API key in Resend.',
    },
    {
        id: MenuState.list,
        label: 'List',
        description: 'List all API keys in Resend.',
    },
    {
        id: MenuState.delete,
        label: 'Delete',
        description: 'Delete an existing API key from Resend.',
    },
];
export const Menu = ({ onExit, onSelect, initialSelectedKey }) => {
    return (_jsx(Layout, { headerText: `${config.baseTitle} - API Keys`, showNavigationInstructions: true, children: _jsx(UIMenu, { menuItems: MAIN_MENU_ITEMS, onSelect: (menuId) => {
                onSelect(menuId);
            }, onExit: () => {
                onExit();
            }, initialSelectedKey: initialSelectedKey }) }));
};
