// @letanure/resend-cli/dist/src/modules/domains/Menu.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Layout } from '../../components/ui/layout.js';
import { Menu as UIMenu } from '../../components/ui/Menu.js';
import { config } from '../../config/config.js';
export var MenuState;
(function (MenuState) {
    MenuState["create"] = "create";
    MenuState["retrieve"] = "retrieve";
    MenuState["verify"] = "verify";
    MenuState["update"] = "update";
    MenuState["list"] = "list";
    MenuState["delete"] = "delete";
})(MenuState || (MenuState = {}));
const DOMAINS_MENU_ITEMS = [
    {
        id: MenuState.create,
        label: 'Create',
        description: 'Create a domain through the Resend Email API.',
    },
    {
        id: MenuState.retrieve,
        label: 'Retrieve',
        description: 'Retrieve a single domain for the authenticated user.',
    },
    {
        id: MenuState.verify,
        label: 'Verify',
        description: 'Verify an existing domain.',
    },
    {
        id: MenuState.update,
        label: 'Update',
        description: 'Update an existing domain.',
    },
    {
        id: MenuState.list,
        label: 'List',
        description: 'Retrieve a list of domains for the authenticated user.',
    },
    {
        id: MenuState.delete,
        label: 'Delete',
        description: 'Remove an existing domain.',
    },
];
export const Menu = ({ onExit, onSelect, initialSelectedKey }) => {
    return (_jsx(Layout, { headerText: `${config.baseTitle} - Domains`, showNavigationInstructions: true, children: _jsx(UIMenu, { menuItems: DOMAINS_MENU_ITEMS, onSelect: (menuId) => {
                onSelect(menuId);
            }, onExit: () => {
                onExit();
            }, initialSelectedKey: initialSelectedKey }) }));
};
