// @letanure/resend-cli/dist/src/modules/broadcasts/list/Form.js
import { jsx as _jsx } from "react/jsx-runtime";
import { ListDisplay } from '../../../components/ui/ListDisplay.js';
import { listBroadcasts } from './action.js';
import { displayFields } from './fields.js';
export const Form = ({ onExit }) => {
    const formatData = (data) => {
        return data.data.map((broadcast) => ({
            id: broadcast.id,
            name: broadcast.name || '',
            audienceId: broadcast.audience_id || '',
            status: broadcast.status,
            created_at: new Date(broadcast.created_at).toLocaleString(),
            scheduled_at: broadcast.scheduled_at ? new Date(broadcast.scheduled_at).toLocaleString() : '',
            sent_at: broadcast.sent_at ? new Date(broadcast.sent_at).toLocaleString() : '',
        }));
    };
    return (_jsx(ListDisplay, { title: "Broadcasts - List", onExit: onExit, loadFunction: listBroadcasts, displayFields: displayFields, formatData: formatData, loadData: {}, noDataMessage: "No broadcasts found." }));
};
