// @letanure/resend-cli/dist/src/modules/contacts/Menu.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Layout } from '../../components/ui/layout.js';
import { Menu as UIMenu } from '../../components/ui/Menu.js';
import { config } from '../../config/config.js';
export var MenuState;
(function (MenuState) {
    MenuState["create"] = "create";
    MenuState["retrieve"] = "retrieve";
    MenuState["update"] = "update";
    MenuState["delete"] = "delete";
    MenuState["list"] = "list";
})(MenuState || (MenuState = {}));
const CONTACTS_MENU_ITEMS = [
    {
        id: MenuState.create,
        label: 'Create',
        description: 'Create a contact inside an audience.',
    },
    {
        id: MenuState.retrieve,
        label: 'Retrieve',
        description: 'Retrieve a single contact from an audience.',
    },
    {
        id: MenuState.update,
        label: 'Update',
        description: 'Update an existing contact.',
    },
    {
        id: MenuState.delete,
        label: 'Delete',
        description: 'Remove an existing contact from an audience.',
    },
    {
        id: MenuState.list,
        label: 'List',
        description: 'Show all contacts from an audience.',
    },
];
export const Menu = ({ onExit, onSelect, initialSelectedKey }) => {
    return (_jsx(Layout, { headerText: `${config.baseTitle} - Contacts`, showNavigationInstructions: true, children: _jsx(UIMenu, { menuItems: CONTACTS_MENU_ITEMS, onSelect: (menuId) => {
                onSelect(menuId);
            }, onExit: () => {
                onExit();
            }, initialSelectedKey: initialSelectedKey }) }));
};
