// @letanure/resend-cli/dist/src/modules/audiences/Menu.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Layout } from '../../components/ui/layout.js';
import { Menu as UIMenu } from '../../components/ui/Menu.js';
import { config } from '../../config/config.js';
export var MenuState;
(function (MenuState) {
    MenuState["create"] = "create";
    MenuState["retrieve"] = "retrieve";
    MenuState["delete"] = "delete";
    MenuState["list"] = "list";
})(MenuState || (MenuState = {}));
const AUDIENCES_MENU_ITEMS = [
    {
        id: MenuState.create,
        label: 'Create',
        description: 'Create a list of contacts.',
    },
    {
        id: MenuState.retrieve,
        label: 'Retrieve',
        description: 'Retrieve a single audience.',
    },
    {
        id: MenuState.delete,
        label: 'Delete',
        description: 'Remove an existing audience..',
    },
    {
        id: MenuState.list,
        label: 'List',
        description: 'Retrieve a list of audiences..',
    },
];
export const Menu = ({ onExit, onSelect, initialSelectedKey }) => {
    return (_jsx(Layout, { headerText: `${config.baseTitle} - Audiences`, showNavigationInstructions: true, children: _jsx(UIMenu, { menuItems: AUDIENCES_MENU_ITEMS, onSelect: (menuId) => {
                onSelect(menuId);
            }, onExit: () => {
                onExit();
            }, initialSelectedKey: initialSelectedKey }) }));
};
