// @letanure/resend-cli/dist/src/modules/broadcasts/Menu.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Layout } from '../../components/ui/layout.js';
import { Menu as UIMenu } from '../../components/ui/Menu.js';
import { config } from '../../config/config.js';
export var MenuState;
(function (MenuState) {
    MenuState["create"] = "create";
    MenuState["retrieve"] = "retrieve";
    MenuState["update"] = "update";
    MenuState["send"] = "send";
    MenuState["delete"] = "delete";
    MenuState["list"] = "list";
})(MenuState || (MenuState = {}));
const BROADCASTS_MENU_ITEMS = [
    {
        id: MenuState.create,
        label: 'Create',
        description: 'Create a new broadcast to send to your audience.',
    },
    {
        id: MenuState.retrieve,
        label: 'Retrieve',
        description: 'Retrieve a single broadcast..',
    },
    {
        id: MenuState.update,
        label: 'Update',
        description: 'Update a broadcast to send to your audience.',
    },
    {
        id: MenuState.send,
        label: 'Send',
        description: 'Start sending broadcasts to your audience through the Resend API.',
    },
    {
        id: MenuState.delete,
        label: 'Delete',
        description: 'Remove an existing broadcast.',
    },
    {
        id: MenuState.list,
        label: 'List',
        description: 'Retrieve a list of broadcast.',
    },
];
export const Menu = ({ onExit, onSelect, initialSelectedKey }) => {
    return (_jsx(Layout, { headerText: `${config.baseTitle} - Broadcasts`, showNavigationInstructions: true, children: _jsx(UIMenu, { menuItems: BROADCASTS_MENU_ITEMS, onSelect: (menuId) => {
                onSelect(menuId);
            }, onExit: () => {
                onExit();
            }, initialSelectedKey: initialSelectedKey }) }));
};
